oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/QueryEngineSettingsMBean.java

```diff
@@ -86,6 +86,24 @@
 
     void setFastQuerySize(boolean fastQuerySize);
 
+    /**
+     * Whether Path restrictions are enabled while figuring out index plan
+     *
+     * @return true if enabled
+     */
+    boolean isEnablePathRestrictions();
```

Comment:
Maybe it's a bit clearer when we use "get / setStrictPathRestrictions"?

Reply:
Right now this is a boolean setting: if disabled, ignore the path condition, if enabled, use it. But we need to do it two stages. The first stage is, only log a warning if indexes are used where the query path condition doesn't match the index path restriction. So 3 stages:

- disabled (current behavior)
- warn (log a warning only, but still use those indexes)
- enabled (in the (more) far future)

The "enabled" behavior is implemented in the cost estimation phase (like now). The "warn" behaviour would need to be implemented when _executing_ a query.

Reply:
The question is, should we have an "enum" instead of a boolean flag? Or two boolean flags. I would opt for an "enum" setting: "get / setStrictPathRestrictions": DISABLED, WARN, ENABLED.